relayer/chains/cosmos/tx.go

```diff
@@ -1086,6 +1086,8 @@ func (cc *CosmosProvider) NewClientState(
 
 func (cc *CosmosProvider) UpdateFeesSpent(chain, key string, fees sdk.Coins) {
 	// Don't set the metrics in testing
+	cc.metricsMu.Lock()
+	defer cc.metricsMu.Unlock()
```

Comment:
If cc.metrics never changes after initialization, then this mutex can have a narrower scope to only protect `cc.TotalFees`.

Reply:
done